packages/react-instantsearch/src/components/ClearAll.js

```diff
@@ -10,13 +10,11 @@ class ClearAll extends Component {
     translate: PropTypes.func.isRequired,
     items: PropTypes.arrayOf(PropTypes.object).isRequired,
     refine: PropTypes.func.isRequired,
-    query: PropTypes.string,
   };
 
   render() {
-    const {translate, items, refine, query} = this.props;
```

Comment:
If I remember correctly this is a part when we need to decide if we go for a breaking change or not. It's also related to comment below (about currentRefinements and weird comment :d) 

Basically now queries are present under items (filtered when no usage of clearsQuery). With the introduction of MultiIndex, then you can have several queries at the same time. Thus this "single" query has a meaning only if use with our regular mode. 

Basically I'm in favor for a breaking change here, as it would be cleaner. It will impact those who were not using the "clearAll" widget with the "clearsQuery" props, but redo their own with the connector.